File: src/types/scan.ts
export type FileCategory =
    | "Video"
    | "Image"
    | "Audio"
    | "Document"
    | "Archive"
    | "Code"
    | "Executable"
    | "System"
    | "Database"
    | "Font"
    | "Other";

export interface ScanNode {
    name: string;
    path: string;
    size: number;
    is_dir: boolean;
    children: ScanNode[];
    file_count: number;
    category: FileCategory;
    last_modified: number | null;
}

export interface ScanProgress {
    files_scanned: number;
    bytes_scanned: number;
    current_path: string;
    percent: number;
    elapsed_ms: number;
}

File: src-tauri/src/models/scan.rs
use super::file::FileCategory;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ScanNode {
    pub name: String,
    pub path: String,
    pub size: u64,
    pub is_dir: bool,
    pub children: Vec<ScanNode>,
    pub file_count: u64,
    pub category: FileCategory,
    pub last_modified: Option<u64>,
}
